src/qr_login.rs

```diff
@@ -0,0 +1,140 @@
+//! Types for QR code login
+
+use matrix_sdk_crypto::types::qr_login;
+use url::Url;
+use wasm_bindgen::prelude::*;
+
+use crate::vodozemac::Curve25519PublicKey;
+
+/// The mode of the QR code login.
+///
+/// The QR code login mechanism supports both, the new device, as well as the
+/// existing device to display the QR code.
+///
+/// The different modes have an explicit one-byte identifier which gets added to
+/// the QR code data.
+#[wasm_bindgen]
+#[derive(Debug)]
+pub enum QrCodeMode {
+    /// Enum variant for the case where the new device is displaying the QR
+    /// code.
+    Login,
+    /// Enum variant for the case where the existing device is displaying the QR
+    /// code.
+    Reciprocate,
```

Comment:
I find these names a bit confusing, but maybe they're referring to names used in the spec? If not, could we consider giving them some names that are more eloquent, like `QrCodeMode` => `QrCodeLoginOrigin` (or s/origin/source), `Login` => `NewDevice`, `Reciprocate` => `ExistingDevice`?

Independently of the above question: it's weird to me to have `Enum variant for the case where` in the doc comment, since this applies to all the enum variants in the world.  Could we get rid of that please?

Reply:
The terms are from the MSC: https://github.com/matrix-org/matrix-spec-proposals/blob/element-hq/oidc-qr-login/proposals/4108-oidc-qr-login.md#qr-code-format. I did shorten them from "initiate login" and "reciprocate login" to "Login" and "Reciprocate" though. I can use the longer names here but I would stick to the names used in the spec.

The docs on the enum variants were shortened in e922f0813877d4841c138585bd2cb01fa462437f.